core/src/main/java/org/bitcoinj/wallet/BasicKeyChain.java

```diff
@@ -204,18 +204,20 @@ public void importKey(ECKey key) {
     }
 
     public ECKey findKeyFromPubHash(byte[] pubkeyHash) {
+        ByteString keyHashString = ByteString.copyFrom(pubkeyHash);
         lock.lock();
         try {
-            return hashToKeys.get(ByteString.copyFrom(pubkeyHash));
+            return hashToKeys.get(keyHashString);
```

Comment:
I think it is safe to remove ByteString from the whole class. The class never exposes the hashmap keys and when putting them into the hashmap, it can create a copy of the byte[]s for immutability.